### PR TITLE
rime-zhwiki: init at 20240909

### PR DESCRIPTION
**Pull request by xddxdd**
RIME dictionary file for word entries from zh.wikipedia.org

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ri/rime-zhwiki/package.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  fetchurl,
+  stdenvNoCC,
+  lib,
+}:
+stdenvNoCC.mkDerivation rec {
```
Comment (Guanran928) [inline pkgs/by-name/ri/rime-zhwiki/package.nix:6]:
```suggestion
stdenvNoCC.mkDerivation {
```

File: pkgs/by-name/ri/rime-zhwiki/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  pname = "rime-zhwiki";
+  version = "20240426";
+  src = fetchurl {
+    url = "https://github.com/felixonmars/fcitx5-pinyin-zhwiki/releases/download/0.2.4/zhwiki-20240426.dict.yaml";
```
Comment (jian-lin) [inline pkgs/by-name/ri/rime-zhwiki/package.nix:10]: Newer versions are available.  This does not block the merge of this PR.

File: pkgs/by-name/ri/rime-zhwiki/package.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    cp $src $out/share/rime-data/zhwiki.dict.yaml
+
+    runHook postInstall
+  '';
+
+  meta = {
+    maintainers = with lib.maintainers; [ xddxdd ];
+    description = "RIME dictionary file for entries from zh.wikipedia.org";
+    homepage = "https://github.com/felixonmars/fcitx5-pinyin-zhwiki";
+    license = lib.licenses.unlicense;
```
Comment (jian-lin) [inline pkgs/by-name/ri/rime-zhwiki/package.nix:28]: Wrong license.

https://github.com/felixonmars/fcitx5-pinyin-zhwiki/blob/8d1a0262b894ca194014308f406c19d0153b5b13/README#L44

https://archlinux.org/packages/extra/any/rime-pinyin-zhwiki/

**Review by jian-lin — CHANGES_REQUESTED** [review]

**Review by jian-lin — APPROVED** [review]
It would be good to add an update script in the future.

**Pull request merged**